qiskit/algorithms/amplitude_estimators/ae.py

```diff
@@ -79,27 +86,67 @@ def __init__(
         super().__init__()
 
         # set quantum instance
-        self.quantum_instance = quantum_instance
+        if quantum_instance is not None:
+            warnings.warn(
+                "The quantum_instance argument has been superseded by the sampler argument. "
+                "This argument will be deprecated in a future release and subsequently "
+                "removed after that.",
+                category=PendingDeprecationWarning,
+            )
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
+            self.quantum_instance = quantum_instance
 
         # get parameters
         self._m = num_eval_qubits  # pylint: disable=invalid-name
         self._M = 2**num_eval_qubits  # pylint: disable=invalid-name
 
         self._iqft = iqft
         self._pec = phase_estimation_circuit
+        self._sampler = sampler
+
+    @property
+    def sampler(self) -> BaseSampler | None:
+        """Get the sampler primitive.
+
+        Returns:
+            The sampler primitive to evaluate the circuits.
+        """
+        return self._sampler
+
+    @sampler.setter
+    def sampler(self, sampler: BaseSampler) -> None:
+        """Set sampler primitive.
+
+        Args:
+            sampler: A sampler primitive to evaluate the circuits.
+        """
+        self._sampler = sampler
 
     @property
-    def quantum_instance(self) -> Optional[QuantumInstance]:
-        """Get the quantum instance.
+    @deprecate_function(
+        "The AmplitudeEstimation.quantum_instance getter is pending deprecation. "
+        "This property will be deprecated in a future release and subsequently "
+        "removed after that.",
+        category=PendingDeprecationWarning,
+    )
+    def quantum_instance(self) -> QuantumInstance | None:
+        """Pending deprecation: Get the quantum instance.
```

Comment:
```suggestion
        """Pending deprecation; Get the quantum instance.
```
Can I would suggest changing `:` to `;` here - I tried escaping it r"\:" in Grover and other things. I settled on ; in the end so it all came into the one sentence in the html. With a : it gets interpreted as a type here. Maybe there's another way not sure - as I said I ended up with ; for Grover PR. Here is how it comes out at present

![image](https://user-images.githubusercontent.com/40241007/192049374-f49a8fa3-9b23-4ee9-b7f8-450bd64474b1.png)

This needs doing in a few places, I am just noting this one.